smt.go

```diff
@@ -408,6 +419,7 @@ func (smt *SparseMerkleTree) ProveForRoot(key []byte, root []byte) (SparseMerkle
 	proof := SparseMerkleProof{
 		SideNodes:             nonEmptySideNodes,
 		NonMembershipLeafData: nonMembershipLeafData,
+		SiblingData:           siblingData,
```

Comment:
There should probably be a separate function like ProveForRootUpdatable or similar, that includes the sibling data. Most use cases won't need this sibling data in their proofs.